agent/command/git.go

```diff
@@ -112,6 +116,8 @@ func validateCloneMethod(method string) error {
 
 func (opts cloneOpts) validate() error {
 	catcher := grip.NewBasicCatcher()
+	catcher.NewWhen(opts.useScalar && opts.cloneDepth > 0, "cannot use scalar with clone depth")
+	catcher.NewWhen(opts.useScalar && opts.recurseSubmodules, "cannot use scalar with recurse submodules")
```

Comment:
Also I think this is missing shallowClone and useScalar but I'm not sure. You do it in the new gitFetchProject validate function